tactics/class_tactics.ml

```diff
@@ -931,12 +931,14 @@ module Search = struct
            top_sort evm goals
          else Evar.Set.elements goals
        in
-       let tac = tac <*> Proofview.Unsafe.tclGETGOALS >>=
+       let goalsl = List.map Proofview_monad.with_empty_state goalsl in
+       let tac =
+         Proofview.Unsafe.tclNEWGOALS goalsl <*>
+       tac <*> Proofview.Unsafe.tclGETGOALS >>=
```

Comment:
Indentation is strange in this chunk

Reply:
Right, but it was already. Do you want me to reindent the whole thing?

Reply:
Ah no, sorry, it is only a local problem indeed. Will fix.